hedera-mirror-rest/transactions.js

```diff
@@ -830,6 +837,48 @@ const extractSqlFromTransactionsByIdOrHashRequest = async (transactionIdOrHash,
   return {query: getTransactionQuery(mainConditions.join(' and '), commonConditions.join(' and ')), params};
 };
 
+function isTransactionSuccessful(result) {
+  return (
+    result === successTransactionResult ||
+    result === successFeeSchedulePartUploadedTransactionResult ||
+    result === successMissingOperationTransactionResult
+  );
+}
+
+function getTransactionsByIdOrHashCacheControlHeader(transactions, transactionIdOrHash, filters) {
+  let hasScheduleCreate = false;
+  let scheduleExecuted = false;
+  const currSecs = Math.floor(new Date().getTime() / 1000);
+  let timestamp = 0;
+  if (filters.some((f) => f.key === constants.filterKeys.SCHEDULED) || isValidTransactionHash(transactionIdOrHash)) {
+    return {}; // no override
+  }
```

Comment:
If transaction hash or scheduled=true then there will be only one result in the list so we can simplify.

```suggestion
  if (transactions.length < 2 || filters.some((f) => f.key === constants.filterKeys.SCHEDULED)) {
    return {}; // no override
  }
```